src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -855,6 +872,76 @@ func validateListener(
 				external.Subdomain,
 				"TLS requires specifying a subdomain"))
 	}
+	if tlsEnabled && clientCACertRef != nil {
+		if !requireClientAuth {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("requireClientAuth"),
+					requireClientAuth,
+					"Enabled has to be set to true for RequireClientAuth if ClientCACertRef is set"))
+		}
+
+		if clientCACertRef.Name == "" {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("clientCACertRef"),
+					clientCACertRef,
+					"Name must be provided if ClientCACertRef is set"))
+		}
+
+		if clientCACertRef.Kind != "" && !strings.EqualFold(clientCACertRef.Kind, "Secret") {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("clientCACertRef"),
+					clientCACertRef,
+					"Kind must be set to secret if set in ClientCACertRef"))
```

Comment:
There's more than one Kind in the cluster resource. What do you think about being a little more specific?

```suggestion
					"clientCACertRef.Kind must be set to secret if set in ClientCACertRef"))
```

Reply:
Same question for Enabled, Name, etc.